src/_gettsim/wohngeld/miete.py

```diff
@@ -172,27 +197,27 @@ def min_miete_m_hh(
     start_date="1984-01-01",
     end_date="2008-12-31",
     leaf_name="miete_m_hh",
-    vectorization_strategy="loop",
 )
 def miete_m_hh_mit_baujahr(
     mietstufe: int,
     wohnen__baujahr_immobilie_hh: int,
     anzahl_personen_hh: int,
     wohnen__bruttokaltmiete_m_hh: float,
     min_miete_m_hh: float,
-    max_miete_m_lookup: dict[int, ConsecutiveInt2dLookupTableParamValue],
+    max_miete_m_lookup: LookupTableBaujahr,
 ) -> float:
     """Rent considered in housing benefit calculation on household level until 2008."""
+
     selected_bin_index = np.searchsorted(
-        np.asarray(sorted(max_miete_m_lookup.keys())),
+        max_miete_m_lookup.baujahre,
         wohnen__baujahr_immobilie_hh,
         side="left",
     )
-    constr_year = list(max_miete_m_lookup.keys())[selected_bin_index]
-    lookup = max_miete_m_lookup[constr_year]
-    max_miete_m = lookup.values_to_look_up[
-        anzahl_personen_hh - lookup.base_to_subtract_rows,
-        mietstufe - lookup.base_to_subtract_cols,
+    max_miete_m = max_miete_m_lookup.lookup_table[
+        selected_bin_index,
+        anzahl_personen_hh
```

Comment:
```suggestion
        anzahl_personen_hh - max_miete_m_lookup.lookup_base_to_subtract_rows[selected_bin_index],  # noqa: E501
```
Out of the top of my head, much easier to read despite the length.

Reply:
I could not get this to work with `#noqa`, skipping the formatting worked though.